plugins/kubernetes/app/models/kubernetes/deploy_executor.rb

```diff
@@ -56,19 +58,25 @@ def execute!(*_commands)
 
     private
 
-    # TODO only call once per cluster and filter the output
-    def pod_is_live?(release_doc)
+    def deployed_pods(release)
+      release.release_docs.map(&:deploy_group).flat_map do |deploy_group|
+        query = {
+          namespace: deploy_group.kubernetes_namespace,
+          label_selector: {
+            deploy_group_id: deploy_group.id,
+            project_id: @job.project_id,
+            release_id: release.id
+          }.map { |k, v| "#{k}=#{v}" }.join(',')
+        }
+        deploy_group.kubernetes_cluster.client.get_pods(query)
```

Comment:
Have this return an array of `Kubernetes::Api::Pod` objects, so you can use their helper methods